contributor/04-mechanics-of-contributing.md

```diff
@@ -0,0 +1,125 @@
+It is very likely that as you do this everything will feel more and more naturally - maybe you'll even wonder why you were doing anything else before.
+
+## Preparing to work
+
+### Lead times
+
+One key difference is the turn-around time. 
+You're coming to a new (host) team on every first contribution and thus you'll need to get to know their approach, i.e. their development, reviews and deployment, etc. processes.
+Even if you are coming back after a while the teams' ways and members might have changed - take your time as you would to catch up with a friend you haven't seen in a while and whom you are visiting now.
+
+Thus: Please give yourself enough lead time.
+Start early enough so that your work is available for you to leverage at the time you need it. 
+It's better to add more slack time initially - you'll get a feeling about the turn-around times once you work with the host team.
+If you end with too much there surely will be a great way to make use of the time, be it some work on your side, the quirky bug or improvement opportunity you found while making your initial contribution or possibly even a nice chat over coffee, beers in case of personal availability or whatever topic in case of distributed projects. 
+
+[EDITOR NOTE: There was actually a paper on how trust is build up on remote/asynchronous collaboration. Maybe I'll find that again. 
+The basic point is that aside of reliability in collaboration, continued presence the effect of water cooler discussions and food/drink consumption is quite impressive...
+]  
+
+Especially in the case of larger distances between you and the host teams, e.g. multiple time zones difference as in San Francisco, USA (PDT) and Berlin, Germany (CEST), you will need to actively consider this and explicitly manage your and their expectations on reaction times.
+
+### Avoiding rejection
+
+You've got this really large feature or larger rewrite in mind that would be totally great and add real value that you would like to donate?
+Excellent!
+Wouldn't it be horrible if all your work would be wasted because the host team is maybe already building something very similar, is planning to depreciate the part of the software you are trying to extend or even does not see what you are proposing to be a fit for their project? 
+This challenge is a frequent challenge and many team relationships suffered from not handling the fit in advance.
+
+Make yourself and the host team happy (and possibly even save some work) by getting agreement from the host team on the user/technical design of the contribution _before_ submitting a pull request.
+You'll have to understand how the host team would like you to reach out for this - sometimes they have regular design discussions in person or via chat/voice/video conferences, some hand in design discussion / extension proposal documents for collective discussion and review, some work with feature requests, etc. 
```

Comment:
This is the Apache person in me speaking: I'd be careful with even mentioning that a host team relies on regular design discussions in person or via chat/voice/video conferences. Great Inner Source projects favor written over verbal advise - this should also include design decisions.

I'd split that sentence into multiple pieces:

- How to figure out, where the project is heading (contributing.md should tell you through which channels the host team discusses and decides on design, roadmap etc.)
- Mention that while there may be in person discussion through various channels, contributing.md will tell you where to find summaries of these discussions and which channels to use to participate in these discussions in an asynchronous way.

I admit that training contributors to go look for these resources and teaching them to demand these resources is a hack towards making host teams operate in a more remote friendly, async friendly way. That in turn should help with creating passive documentation of the project direction. It should also help with answering the question of "why did we choose that implementation path" a couple months or years later.

Reply:
You are very right here.
I'll try to rephrase things here. I'd also add that having regular non-written conversations might be totally ok, however that time-zone-based participation limitations need to be considered for inclusivity and that the final goal needs to be having everything written & linkable/searchable.

Side note: an interesting approach are some of the meetings that OpenZFS holds are recorded and put online on Youtube - generating possibly searchable subtitles. Pure text is easier though...